src/palace/manager/integration/license/opds/opds2/api.py

```diff
@@ -70,6 +73,16 @@ def __init__(self, _db: Session, collection: Collection):
         self.token_auth_configuration: str | None = (
             collection.integration_configuration.context.get(self.TOKEN_AUTH_CONFIG_KEY)
         )
+        self._request = get_opds_requests(
+            (
+                OpdsAuthType.BASIC
+                if self.settings.username and self.settings.password
```

Comment:
The password is not strictly required for HTTP Basic auth. Does our implementation require it? I not, we might consider leaving it off in this check.

Additionally, it doesn't feel quite right to switch over to no auth if a username is configured.

```suggestion
                if self.settings.username
```